### PR TITLE
Bugfix/#226 #227 error handling

### LINKED ISSUE #226: error messages are unused
Although the responses from the server (in case of an error) contain specific messages the frontend does not display them.

### LINKED ISSUE #227: Creation of first entity is not properly validated
The validation of the first school year and semester does not work as expected since the refactoring.
Therefore the start date might be bigger than the end date.

This only applys for the first child of each parent.

### PR DESCRIPTION
**Pull request by needToRoll**
This closes #226, closes #227 

Summary of changes:
- ​Added error handling method in HttpInterceptor
- Updated I18N Keys in ValidationKeys
- Fixed semester and school year validations

I made sure to:
- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines
- [x] ~Update CI script and configs on Jenkins (`Jenkinsfile`, `auth0.yml`, `application.yml`, etc.)~

Please review @jmesserli  / cc: @outcobra/developers.

### CODE REVIEW

**Comment by jmesserli** [comment]
The english translations seem to be missing

**Comment by needToRoll** [comment]
Ok thanks, does the ValidationException file belong where it is now or into the frontend/src/app/core/model folder

File: frontend/src/app/core/http/HttpInterceptor.ts [hunk 1]
```diff
@@ -291,5 +290,24 @@ export class HttpInterceptor {
...
+    /**
+     * This function handles errors that occur when making a http-request
+     * If it was a call to our own api and the server responded with a {ValidationException} a specific error message will be displayed.
+     * @param error the error that occurred during the request
+     * @returns {Observable<T>} the Observable containing the error information
+     */
+    private _handleError<T>(error: any): Observable<T> {
+        try {
+            let body = error._body
+            let exception = JSON.parse(body.toString(), dateReviver) as ValidationException
```
Comment (M4R1KU) [inline frontend/src/app/core/http/HttpInterceptor.ts]: Nope nope nope....there's a method for this

File: frontend/src/app/core/http/HttpInterceptor.ts [hunk 2]
```diff
@@ -291,5 +290,24 @@ export class HttpInterceptor {
...
+            this.notificationsService.error(exception.title, exception.message)
+            return Observable.throw(exception)
+        } catch (exception) {
```
Comment (M4R1KU) [inline frontend/src/app/core/http/HttpInterceptor.ts]: Does this display 2 notifications or what does this catch exactly?

File: frontend/src/app/core/http/HttpInterceptor.ts (hunk 1 above)
Comment (M4R1KU) [inline frontend/src/app/core/http/HttpInterceptor.ts]: Beide lines remove und mit `this._unwrapAndCastHttpResponse(error)` ersetze

**Pull request merged**